app/src/interfaces/_system/system-fields/system-fields.vue

```diff
@@ -100,6 +109,19 @@ const fields = computed<(Field & { key: string })[]>({
 
 const { t } = useI18n();
 
+function selectAll() {
+	const newArray = props.value ?? [];
+
+	const fullArray = fieldsStore.getFieldsForCollection(props.collectionName).map((field) => field.field);
+
+	fullArray.map((field) => {
+		if (props.value?.includes(field) === false) {
+			newArray.push(field);
+		}
+	});
+	emit('input', newArray);
+}
```

Comment:
Should we use `.forEach` instead of `.map` here, just in terms of clarifying that it's only used for iterating `fullArray`?

---

Alternatively, maybe we can use Set here? such as:

```js
function selectAll() {
	const fields = fieldsStore.getFieldsForCollection(props.collectionName).map((field) => field.field);
	const allFields = new Set([...(props.value ?? []), ...fields]);
	emit('input', Array.from(allFields));
}
```

since it seems like we're also using Set throughout the app, which a similar example being:

https://github.com/directus/directus/blob/0154878f2746809896738a1cfb73edcaa3dceace/app/src/components/v-checkbox-tree/v-checkbox-tree.vue#L101-L103

Reply:
I've used set as recommended but I am keeping `.map` for this as map works well and `.forEach` had issues handing it, also when querying other times we've cycled through collections fields in core, we usually use `.map` as well.